cmake/developer_package/compile_flags/functions.cmake

```diff
@@ -479,3 +479,38 @@ function(ov_target_link_libraries_as_system TARGET_NAME LINK_TYPE)
         set_target_properties(${TARGET_NAME} PROPERTIES NO_SYSTEM_FROM_IMPORTED ON)
     endif()
 endfunction()
+
+#
+# Detects the standard filesystem library and sets the OPENVINO_STD_FS_LIB variable with library name if required.
+#
+function(ov_detect_std_filesystem_lib)
+    if(NOT DEFINED OPENVINO_STD_FS_LIB)
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+            "#include <experimental/filesystem>\nint main(int argc, char ** argv) {\n  std::experimental::filesystem::path p(argv[0]);\n  return p.string().length();\n}")
+        try_compile(STD_FS_NO_LIB_NEEDED ${CMAKE_CURRENT_BINARY_DIR}
+                    SOURCES ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+                    COMPILE_DEFINITIONS -std=c++11)
+        try_compile(STD_FS_NEEDS_STDCXXFS ${CMAKE_CURRENT_BINARY_DIR}
+                    SOURCES ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+                    COMPILE_DEFINITIONS -std=c++11
+                    LINK_LIBRARIES stdc++fs)
+        try_compile(STD_FS_NEEDS_CXXFS ${CMAKE_CURRENT_BINARY_DIR}
+                    SOURCES ${CMAKE_CURRENT_BINARY_DIR}/stdfs/main.cpp
+                    COMPILE_DEFINITIONS -std=c++11
+                    LINK_LIBRARIES c++fs)
+
+        if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+            message(STATUS "MSVC - No explicit filesystem linker setting required.")
+        elseif(STD_FS_NEEDS_STDCXXFS)
+            message(STATUS "STD_FS_NEEDS_STDCXXFS - Add explicit filesystem linker setting: 'stdc++fs'.")
+            set(OPENVINO_STD_FS_LIB stdc++fs CACHE INTERNAL "Library name for filesystem")
+        elseif(STD_FS_NEEDS_CXXFS)
+            message(STATUS "STD_FS_NEEDS_CXXFS - Add explicit filesystem linker setting: 'c++fs'.")
+            set(OPENVINO_STD_FS_LIB stdc++fs CACHE INTERNAL "Library name for filesystem")
+        elseif(STD_FS_NO_LIB_NEEDED)
+            message(STATUS "STD_FS_NO_LIB_NEEDED - No explicit filesystem linker setting required.")
```

Comment:
Should STD_FS_NO_LIB_NEEDED be the first if check?
I mean if it is compiled without any extra dependency, then we shoud not add anything to OPENVINO_STD_FS_LIB

Reply:
This logic is copied and was working so will keep as it is.
@barnasm1  should be updated?